R/st_bonds.R

```diff
@@ -0,0 +1,194 @@
+#' Sensitivity analysis: `run_stress_test()` multiple times
+#'
+#' These functions performs the so called "sensitivity analysis" for the asset
+#' type in their name. They run [run_stress_test()] multiple times, iterating
+#' over multiple values of one and only one argument of [run_stress_test()].
+#'
+#' @param data Character vector of length 2. Paths to the directories
+#'   ST_INPUTS_MASTER and ST_TESTING_\{aset-type\}, e.g. ST_TESTING_BONDS,
+#'   respectively. These directories must have the files expected by the
+#'   stress-test project. The easiest way to create this object is with
+#'   [st_data_paths()].
+#' @param destdir Character. Path to a destination directory.
+#' @param ... Arguments passed to [run_stress_test()].
+#' @param quiet Logical. Print non-condition messages to the console?
+#'
+#' @seealso [st_data_paths()].
+#' @family main functions
+#'
+#' @return A data frame.
+#'
+#' @export
+#'
+#' @examplesIf r2dii.climate.stress.test:::is_registered_dev("/home/mauro")
+#' library(readr, warn.conflicts = FALSE)
+#'
+#' # Give the paths to your data explicitely
+#' data <- st_data_paths(
+#'   data = "/home/mauro/tmp/st/ST_INPUTS_MASTER",
+#'   project = "/home/mauro/tmp/st/ST_TESTING_BONDS"
+#' )
+#' out <- suppressWarnings(
+#'   st_bonds(data, term = c(1, 2))
+#' )
+#'
+#' # Or set these environment variables (e.g. in .Renviron), then omit `data`
+#' data
+#'
+#' # Here we omit `data` and make the console output more verbose
+#' out <- st_bonds(term = c(1, 2), quiet = FALSE))
+#'
+#' # The data frame output helps you quickly explore and manipulate your results
+#' subset(out, st_name == "port" & arg_value == 2)
+#'
+#' # If you run out of memory, instead write output sequentially to disk
+#' path <- file.path(tempdir(), "st")
+#' dir.create(path)
+#' suppressWarnings(st_write_bonds(destdir = path, term = c(1, 2)))
+#' list.files(path)
+#'
+#' # Later you may read all results at once
+#' files <- list.files(path, full.names = TRUE)
+#' results <- read_csv(files, id = "arg_val", show_col_types = FALSE)
+#' results
+#'
+#' # The column `arg_val` helps you identify each run
+#' unique(
+#'   subset(results, grepl("term___2", arg_val), select = 1:3)
+#' )
+st_bonds <- function(data = st_data_paths(), ..., quiet = TRUE) {
+  st_df(data, asset_type = "bonds", ..., quiet = quiet)
+}
+
+#' @rdname st_bonds
+#' @export
+st_equity <- function(data = st_data_paths(), ..., quiet = TRUE) {
+  st_df(data, asset_type = "equity", ..., quiet = quiet)
+}
+
+#' @rdname st_bonds
+#' @export
+st_loans <- function(data = st_data_paths(), ..., quiet = TRUE) {
+  st_df(data, asset_type = "loans", ..., quiet = quiet)
+}
+
+#' @rdname st_bonds
+#' @export
+st_write_bonds <- function(data = st_data_paths(),
+                           destdir = tempdir(),
+                           ...,
+                           quiet = TRUE) {
+  st_write(data, asset_type = "bonds", destdir = destdir, ..., quiet = quiet)
```

Comment:
It's silly I read and re-write the output, instead of moving it to `destdir`. But for now I'm worried only about the interface and to not touch existing code. Later we can refactor for better performance.